src/number_theory/kummer_dedekind.lean

```diff
@@ -202,46 +205,64 @@ namespace kummer_dedekind
 
 open_locale big_operators polynomial classical
 
-variables [is_domain R] [is_domain S] [is_dedekind_domain S]
-variables (pb : power_basis R S)
+variables [is_domain R] [is_integrally_closed R]
+variables [is_domain S] [is_dedekind_domain S]
+variable [no_zero_smul_divisors R S]
 
 local attribute [instance] ideal.quotient.field
 
 /-- The first half of the **Kummer-Dedekind Theorem** in the monogenic case, stating that the prime
     factors of `I*S` are in bijection with those of the minimal polynomial of the generator of `S`
     over `R`, taken `mod I`.-/
 noncomputable def normalized_factors_map_equiv_normalized_factors_min_poly_mk (hI : is_maximal I)
-  (hI' : I ≠ ⊥) : {J : ideal S | J ∈ normalized_factors (I.map (algebra_map R S) )} ≃
-    {d : (R ⧸ I)[X] | d ∈ normalized_factors (map I^.quotient.mk (minpoly R pb.gen)) } :=
-((normalized_factors_equiv_of_quot_equiv ↑(pb.quotient_equiv_quotient_minpoly_map I)
+  (hI' : I ≠ ⊥) (hx : (conductor R x).comap (algebra_map R S) ⊔ I = ⊤)
+  (hx' : is_integral R x) :
+  {J : ideal S | J ∈ normalized_factors (I.map (algebra_map R S) )} ≃
+    (begin
+      letI : field  (R ⧸ I) := infer_instance,
+      letI : unique_factorization_monoid (R ⧸I)[X] := infer_instance,
+      exact {d : (R ⧸ I)[X] | d ∈ normalized_factors (map I^.quotient.mk (minpoly R x))},
+    end) :=
```

Comment:
You don't need the `letI`s here:
```suggestion
    {d : (R ⧸ I)[X] | d ∈ normalized_factors (map I^.quotient.mk (minpoly R x))} :=
```

Reply:
Yeah I put those to make the code compile faster